### ISSUE TITLE
Stray characters in zactor.c causing build failure (due to -Werror)

### ISSUE THREAD

**Issue by koobs** [zeromq/czmq#1029]
Due to CPPFLAGS defaulting to -Werror via configure:

# These options are GNU compiler specific.
if test "x$GCC" = "xyes"; then
    CPPFLAGS="-pedantic -Werror -Wall -Wc++-compat ${CPPFLAGS}"
fi
```

The following build failure is reproducible on FreeBSD 8.4:

===>  Building for czmq-3.0.2
Making all in doc
/bin/sh ./libtool  --tag=CC   --mode=compile cc -std=gnu99 -DHAVE_CONFIG_H -I. -I./src  -I/usr/local/include  -I./include -pedantic -Werror -Wall -Wc++-compat -D__BSD_VISIBLE -D_REENTRANT -D_THREAD_SAFE  -O2 -pipe -fstack-protector -fno-strict-aliasing -MT src/src_libczmq_la-zactor.lo -MD -MP -MF src/.deps/src_libczmq_la-zactor.Tpo -c -o src/src_libczmq_la-zactor.lo `test -f 'src/zactor.c' || echo './'`src/zactor.c
libtool: compile:  cc -std=gnu99 -DHAVE_CONFIG_H -I. -I./src -I/usr/local/include -I./include -pedantic -Werror -Wall -Wc++-compat -D__BSD_VISIBLE -D_REENTRANT -D_THREAD_SAFE -O2 -pipe -fstack-protector -fno-strict-aliasing -MT src/src_libczmq_la-zactor.lo -MD -MP -MF src/.deps/src_libczmq_la-zactor.Tpo -c src/zactor.c  -fPIC -DPIC -o src/.libs/src_libczmq_la-zactor.o
src/zactor.c:1: error: stray '\357' in program
src/zactor.c:1: error: stray '\273' in program
src/zactor.c:1: error: stray '\277' in program
*** Error code 1
```

Proposing the following changes:

 * Remove spurious characters from zactor.c
 * Remove -Werror from default (release) distribution CPPFLAGS

See: https://blog.flameeyes.eu/2009/02/future-proof-your-code-dont-use-werror for rationale.

**Comment by hintjens** [comment]
Closing this, making specific issues (one issue per problem, please)

**Comment by hintjens** [comment]
Let's handle these issues separately. 

The -Werror issue must be discussed on zeromq-dev because it affects a whole family of projects. I'm not convinced by the arguments for making it optional. Disabling it optionally would work for me.

The BOM header in source files, now that's different. This has to be burnt with fire.

**Comment by koobs** [comment]
Referenced and fixed in #1030 and #1031
